gui/esp.hpp

```diff
@@ -0,0 +1,74 @@
+#pragma once
+
+#include <vector>
+#include <mutex>
+
+#include "state.hpp"
+
+#include <imgui/imgui.h>
+
+static bool IsWithinScreenBounds(Vector2& pos)
+{
+	return pos.x < (float)Screen_get_width(nullptr) && pos.y < (float)Screen_get_height(nullptr);
+}
+
+static ImVec2 WorldToScreen(Vector2 pos)
+{
+	const ImVec2 winsize = { (float)Screen_get_width(nullptr), (float)Screen_get_height(nullptr) };
+
+	// These values are not random, i found them by random testing
+	float scale = winsize.y / 1080.0f;
+	float view = (180 * scale) / (State.CameraHeight / 3); // The value 180 is specific for 1080p so we need to scale it for other resolutions
```

Comment:
Why hardcode this?

Reply:
this esp will work for every res out there. This is the fov or view for the 1080p res so i am scaling off that